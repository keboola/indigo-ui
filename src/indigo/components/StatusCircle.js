import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircle } from '@fortawesome/free-solid-svg-icons/faCircle';
import { faCheckCircle } from '@fortawesome/free-solid-svg-icons/faCheckCircle';
import { faTimesCircle } from '@fortawesome/free-solid-svg-icons/faTimesCircle';
import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons/faExclamationCircle';

const statusMap = {
  success: { status: 'success', icon: faCheckCircle },
  error: { status: 'danger', icon: faExclamationCircle },
  warn: { status: 'danger', icon: faTimesCircle },
  warning: { status: 'danger', icon: faTimesCircle },
  processing: { status: 'warning', icon: faCircle },
  waiting: { status: 'default', icon: faCircle },
  cancelled: { status: 'default', icon: faExclamationCircle },
  terminating: { status: 'default', icon: faExclamationCircle },
  terminated: { status: 'default', icon: faExclamationCircle },
};

const StatusCircle = (props) => {
  const { status = 'default', icon = faCircle } = statusMap[props.status] ?? {};
  return (
    <FontAwesomeIcon
      icon={icon}
      className={classnames(`status-circle-${status}`, props.className)}
    />
  );
};

StatusCircle.propTypes = {
  status: PropTypes.string,
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

export default StatusCircle;
