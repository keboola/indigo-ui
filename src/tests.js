import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
