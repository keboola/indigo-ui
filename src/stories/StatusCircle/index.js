import React from 'react';
import { storiesOf } from '@storybook/react';

import StatusCircle from '../../indigo/components/StatusCircle';

storiesOf('StatusCircle', module)
  .add('Status - Success', () => <StatusCircle status="success" />)
  .add('Status - Error', () => <StatusCircle status="error" />)
  .add('Status - Warning', () => <StatusCircle status="warning" />)
  .add('Status - Processing', () => <StatusCircle status="processing" />)
  .add('Status - Terminated', () => <StatusCircle status="terminated" />)
  .add('Status - Cancelled', () => <StatusCircle status="cancelled" />);
