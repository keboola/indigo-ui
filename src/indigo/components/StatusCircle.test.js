import React from 'react';

import { matchesSnapshot } from '../../tests';
import StatusCircle from './StatusCircle';

describe('<StatusCircle />', () => {
  it('Status - Cancelled', () => {
    matchesSnapshot(<StatusCircle status="cancelled" />);
  });

  it('Status - Success', () => {
    matchesSnapshot(<StatusCircle status="success" />);
  });

  it('Status - Error', () => {
    matchesSnapshot(<StatusCircle status="error" />);
  });

  it('Status - Warning', () => {
    matchesSnapshot(<StatusCircle status="processing" />);
  });

  it('Status - Terminated', () => {
    matchesSnapshot(<StatusCircle status="terminated" />);
  });

  it('Status - not provided', () => {
    matchesSnapshot(<StatusCircle />);
  });

  it('Status - random status', () => {
    matchesSnapshot(<StatusCircle status="random-status" />);
  });
});
