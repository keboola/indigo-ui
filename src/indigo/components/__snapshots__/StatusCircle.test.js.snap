// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusCircle /> Status - Cancelled 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-exclamation-circle fa-w-16 status-circle-default"
  data-icon="exclamation-circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<StatusCircle /> Status - Error 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-exclamation-circle fa-w-16 status-circle-danger"
  data-icon="exclamation-circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<StatusCircle /> Status - Success 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-check-circle fa-w-16 status-circle-success"
  data-icon="check-circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<StatusCircle /> Status - Terminated 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-exclamation-circle fa-w-16 status-circle-default"
  data-icon="exclamation-circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<StatusCircle /> Status - Warning 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 status-circle-warning"
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<StatusCircle /> Status - not provided 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 status-circle-default"
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<StatusCircle /> Status - random status 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 status-circle-default"
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;
